packages/web-forms/src/components/controls/RankControl.vue

```diff
@@ -0,0 +1,209 @@
+<script setup lang="ts">
+import { inject, ref, watch } from 'vue';
+import { VueDraggable } from 'vue-draggable-plus';
+import type { RankItem, RankNode } from '@getodk/xforms-engine';
+import ControlText from '@/components/ControlText.vue';
+import ValidationMessage from '@/components/ValidationMessage.vue';
+
+interface RankControlProps {
+	readonly question: RankNode;
+}
+
+interface RankDraggableOption {
+	value: string;
+	label: string;
+}
+
+const props = defineProps<RankControlProps>();
+const HOLD_DELAY = 60; // Delay in ms to hold an item before dragging, avoids accidental reordering on swipe.
+const options = ref<RankDraggableOption[]>([]);
+const touched = ref(false);
+const submitPressed = inject<boolean>('submitPressed');
+const highlight = {
+	index: ref(null),
+	timeoutID: null,
+};
+
+const transformOptions = (rankOptions: RankItem[]) => {
+	if (options.value.length) {
+		options.value.forEach((option) => {
+			option.label = props.question.getValueLabel(option.value);
+		});
+		return;
+	}
+
+	options.value = rankOptions.map((option) => {
+		return { value: option.value, label: props.question.getValueLabel(option.value) };
+	});
+};
+
+watch(props.question.currentState.valueOptions, transformOptions, { immediate: true });
+
+const setValues = () => {
+	touched.value = true;
+	props.question.setValues(options.value.map((option) => option.value));
+};
+
+const setHighlight = (index: number) => {
+	highlight.index.value = index;
+
+	if (highlight.timeoutID) {
+		clearTimeout(highlight.timeoutID);
+		highlight.timeoutID = null;
+	}
+
+	if (highlight.index.value !== null) {
+		highlight.timeoutID = setTimeout(() => setHighlight(null), 1000);
+	}
+};
+
+const moveUp = (index: number) => {
+	const newPosition = index - 1;
+	if (newPosition < 0) {
+		return;
+	}
+	swapItems(index, newPosition);
+};
+
+const moveDown = (index: number) => {
+	const newPosition = index + 1;
+	if (newPosition >= options.value.length) {
+		return;
+	}
+	swapItems(index, newPosition);
+};
+
+const swapItems = (index: number, newPosition: number) => {
+	setHighlight(index);
+	const temp = options.value[index];
+	options.value[index] = options.value[newPosition];
+	options.value[newPosition] = temp;
+	setHighlight(newPosition);
+};
+</script>
+
+<template>
+	<ControlText :question="question" />
+
+	<VueDraggable
+		:id="question.nodeId"
+		v-model="options"
+		:delay="HOLD_DELAY"
```

Comment:
```suggestion
		:delay="HOLD_DELAY"
		:delay-on-touch-only="true"
```

It took me several tries in the UI to get an item to drag, before I realized there was a delay. I think there are other ways to make that more obvious without eliminating the delay, but I think intuitively I just don't expect a delay with a mouse. I was pleased to see this option already existed (and of course we're at the mercy of the third party library getting it right, but I think that's a reasonable starting point).

This one change would address _by far_ the biggest concern I have from playing around with the change locally.

Reply:
Oh... I didn't know where to put this, and then forgot to post it in the last set of comments.

I'd like to discuss (and probably bring in @alyblenkin):

- Does it make sense for the whole item to be draggable, or just (a touch-friendly area around) the handle icon?
- Should that differ on touch vs pointer input?
- If we reduce draggable area for pointer input, I'd want to make sure the drag cursor is only shown where dragging works. (This was my original feedback, before I realized that the whole item is draggable, hence the other questions.)

Reply:
> Does it make sense for the whole item to be draggable, or just (a touch-friendly area around) the handle icon?

Last time I met @alyblenkin, it's the whole item, so to make it more accessible, I tried to capture it in [this comment](https://github.com/getodk/web-forms/issues/243#issuecomment-2594192464).

> Hold and drag are anywhere in the option (not just the icon), making them easier to use.

Reply:
> It took me several tries in the UI to get an item to drag, before I realized there was a delay. I think there are other ways to make that more obvious without eliminating the delay, but I think intuitively I just don't expect a delay with a mouse.

I'm going to experiment and see if there's a way to detect touch instead of mouse.

Reply:
> I'm going to experiment and see if there's a way to detect touch instead of mouse.

I was wondering if it was a good idea to mix code suggestions and text the way I have in several comments. In this case I don't think it helped! So just in case you didn't see it in the suggestion, adding `:delay-on-touch-only="true"` should work. It did for me (although I guess I didn't try it on mobile, I'm just assuming the library works!).

It's listed as `delayOnTouchOnly`[^1] in the [`vue-draggable-plus` API docs](https://vue-draggable-plus.pages.dev/en/api/#api-1)—but the docs are wrong about the type, it is definitely a `boolean` in the prop types.

[^1]: It's still taking me some getting used to that Vue uses camelCase and kebab-case interchangeably for props. Just mentioning that here in case it also helps you to keep in mind when referencing Vue-related docs.

Reply:
Oh yeah, I missed it! It's working good in my Android phone as well

Reply:
Resolving as it was discussed in today's team meeting, we are leaving the whole row draggable and observing user feedback